### PR TITLE
Ensure autoHeader dom result is similar to parsed H1

### PR DESCRIPTION
**Pull request by cyrilf**
fix https://github.com/docsifyjs/docsify/issues/516

Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you are merging your commits to `master` branch.
* [x] Add some descriptions and refer relative issues for you PR.
* [x] DO NOT include files inside `lib` directory.

----

In reference to #516 
Ensure the `h1` dom element created by the `autoHeader` feature is similar to the `ħ1` parsed from a markdown file.

```diff
-<h1>Title</h1>
+<h1 id="title">
+  <a href="#/?id=title" data-id="title" class="anchor">
+    <span>Title</span>
+  </a>
+</h1>
```

Please review this code carefully as I don't know if exposing the `renderer` is a good idea.

### CODE REVIEW

File: src/core/render/index.js [hunk 1]
```diff
@@ -115,9 +115,9 @@ export function renderMixin(proto) {
       const main = dom.getNode('#main')
       const firstNode = main.children[0]
       if (firstNode && firstNode.tagName !== 'H1') {
-        const h1 = dom.create('h1')
-        h1.innerText = activeEl.innerText
-        dom.before(main, h1)
+        const h1 = this.compiler.header(activeEl.innerText, 1)
+        const wrapper = dom.create('div', h1)
+        dom.before(main, wrapper.children[0])
```
Comment (cyrilf) [inline src/core/render/index.js]: Because `h1` is returned as a `String`, I use this trick to convert it into a DOM node

**Comment by timaschew** [comment]
Please run these commands to trigger netlify deployment for this Pull Request:
```sh
git commit --amend --no-edit
git push --force
```

**Comment by cyrilf** [comment]
Done. :heavy_check_mark:

**Comment by timaschew** [comment]
@jhildenbiddle  what do you think?

**Comment by jhildenbiddle** [comment]
Is there a preview for this PR? I don't see a link...

**Comment by timaschew** [comment]
Oh, it seems that it’s visible just to me.
https://deploy-preview-811--docsifyjs.netlify.com/#/

I will try to find out how to make them public. And will also add you as a team member in netlify.

https://deploy-preview-811--docsifyjs.netlify.com/#/

**Comment by timaschew** [comment]
I don’t think it’s bad to expose the renderer.

**Comment by cyrilf** [comment]
Any update on this PR?
Just received the notification from the bot but I still think it's a nice to have for more consistency. (haven't check master recently tho' so I don't know if it's still relevant :see_no_evil: )

**Comment by anikethsaha** [comment]
Please rebase.

File: src/core/render/compiler.js [hunk 2]
```diff
@@ -80,7 +80,7 @@ export class Compiler {
     this.linkTarget = config.externalLinkTarget || '_blank'
     this.contentBase = router.getBasePath()
 
-    const renderer = this._initRenderer()
+    const renderer = this.renderer = this._initRenderer()
```
Comment (anikethsaha) [inline src/core/render/compiler.js]: exposing just the `heading` method seems better than the renderer

**Comment by cyrilf** [comment]
Rebased and updated your requested change :v:

**Pull request merged**